docs/source/installation/guide_complete/index.rst

```diff
@@ -0,0 +1,363 @@
+.. _installation:guide-complete:
+
+===========================
+Complete installation guide
+===========================
+
+The :ref:`quick installation guide <installation:guide-quick>` is designed to make the setup as simple and portable as possible.
+However, the resulting setup has some :ref:`limitations <installation:guide-quick:limitations>` concerning the available functionality and performance.
+This guide provides detailed information and instructions to set up a feature-complete and performant installation.
+
+To get a working installation of AiiDA, the following steps are required:
+
+#. :ref:`Install the Python Package <installation:guide-complete:python-package>`
+#. :ref:`Create a profile <installation:guide-complete:create-profile>`
+
+
+.. _installation:guide-complete:python-package:
+
+Install Python Package
+======================
+
+.. important::
+    AiiDA requires a recent version of Python.
+    Please refer to the `Python Package Index (PyPI) <https://pypi.org/project/aiida-core/>`_ for the minimum required version.
+
+To install AiiDA, the ``aiida-core`` Python package needs to be installed which can be done in a number of ways:
+
+
+.. tab-set::
+
+    .. tab-item:: pip
+
+        Installing ``aiida-core`` from PyPI.
+
+        #. Install `pip <https://pip.pypa.io/en/stable/installation/>`_.
+        #. Install ``aiida-core``:
+
+           .. code-block:: console
+
+               pip install aiida-core
+
+    .. tab-item:: conda
```

Comment:
We don't tell users anymore about the option of installing the services via `conda`? Should we add this here again or in the relevant sections about the services?

Reply:
This was another problem point that I don't know how best to handle. The old approach of the docs was a "vertical" narrative, where you chose an installation "method" e.g., system installation, or conda, or docker, and it went through it from start to finish.

I don't like this approach as you end up having to write out the same things in each vertical option, things like `verdi profile setup`, checking the result with `verdi status` etc. But worse, I feel that users don't get an idea of _what_ exactly is required for AiiDA and _why_. It would make it impossible for them to then get away from any of the examples provided by us and come up with a custom installation.

In this rewrite, I have favored the "horizontal" approach, if you will, where I focus more on the abstract what an AiiDA installation includes, and then it leaves more open as to how those parts are actually performed by the user. For example, when we say that the `core.psql_dos` plugin requires PostgreSQL, it is only there that we then tell the user "ok, you now figure out how you want to use PostgreSQL. That may be through installation on your machine (which again has many options) or you can use a service on another machine".

I just feel that it is untenable for us to map out all the possible options that can occur for different operating systems. Before, we were pretty much forced to provide some of these instructions, because without PSQL, you simply couldn't run AiiDA. However now, it is truly optional, so I think it is fine that we no longer have to provide a gazillion different possibilities of how to install it. It made the original docs unreadable in my opinion.

If you can find a nice way to include the instructions for the conda way, into the new docs outline, that doesn't mess with the main flow and keeps it readable if people don't use conda, I would definitely consider keeping it. But there have historically been a lot of issues with people using the conda-based setup, mostly because the services aren't automatically started upon reboot and the services can clash with equivalents that are running system wide.

Reply:
Hi @sphuber, thanks for the detailed write-up! I definitely see the problems with the previous vertical approach. I just skimmed through those again, and the flow is certainly hard to follow, always having tabs for different options (e.g. OSs, pip+venv / conda, etc.). The new version reads much smoother. 

> I just feel that it is untenable for us to map out all the possible options that can occur for different operating systems. Before, we were pretty much forced to provide some of these instructions, because without PSQL, you simply couldn't run AiiDA. However now, it is truly optional, so I think it is fine that we no longer have to provide a gazillion different possibilities of how to install it. It made the original docs unreadable in my opinion.

Yeah, I agree with that. Why don't we just add an `Info` box in which we link the installation instructions on the RMQ and PSQL webpages, similar to the "Other" tab for the prerequisites, and also mention that the `aiida-core.services` `conda` package exists, so people can alternatively install everything via `conda`. This provides at least some minimal information for the users without cluttering the flow too much? The expected possible issues with `conda` could then be noted in the troubleshooting section, in which we don't have to care about a nice text flow, if we want to mention them at all?

![image](https://github.com/aiidateam/aiida-core/assets/41700727/8a46c3cc-b127-44b9-9854-8a13f31f62a0)